### PR TITLE
Cert resolver

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
How to test:

- `npm i -g blackbaud/skyux-sdk-cli#cert-resolver`
- `skyux certs install --checkout initial-commit` (until https://github.com/blackbaud/skyux-sdk-certs/pull/1)
- Create a new SPA but change the builder reference to `blackbaud/skyux-sdk-builder#cert-resolver` (until https://github.com/blackbaud/skyux-sdk-builder/pull/164)
- Run `skyux serve`

[Documentation PR](https://github.com/blackbaud/skyux2-docs/pull/691)

### CODE REVIEW

File: lib/certs.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+const os = require('os');
+const path = require('path');
+const fs = require('fs-extra');
+const logger = require('@blackbaud/skyux-logger');
+const npmInstall = require('./npm-install');
+const cloneUtil = require('./utils/clone');
+
+function getCertRootPath() {
+  return path.resolve(`${os.homedir()}/.skyux/certs/`);
```
Comment (Blackbaud-SteveBrush) [inline lib/certs.js]: Any reason this value can't be a `const`?

File: lib/certs.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    await fs.remove(certRootPath);
+  } else {
+    logger.info(`Previous clone of certs not found at ${certRootPath}`);
+  }
+}
+
+async function clone() {
+  const certRootPath = getCertRootPath();
+  await cloneUtil('https://github.com/blackbaud/skyux-sdk-certs.git', certRootPath, argv);
+  await npmInstall({ path: certRootPath });
```
Comment (Blackbaud-SteveBrush) [inline lib/certs.js]: This line jumped out at me, but I'm not sure what it's doing?

File: lib/new.js [hunk 3]
```diff
@@ -302,11 +294,11 @@ function getTemplateFromArgs(args) {
   return template;
 }
 
-async function skyuxNew(args) {
+async function skyuxNew(argv) {
```
Comment (Blackbaud-SteveBrush) [inline lib/new.js]: Good change!

File: lib/certs.js (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply lib/certs.js]: Just confirming, you're talking about the `npmInstall` line?  That's simply running `npm install` in the context of the cloned `skyux-sdk-certs` directory.  Same technique and method we use in the `skyux new` functionality.

File: lib/certs.js (hunk 1 above)
Comment (Blackbaud-BobbyEarl) [reply lib/certs.js]: Reworked and this is now just a const.

**Comment by Blackbaud-BobbyEarl** [comment]
Made a few changes based on your feedback @Blackbaud-SteveBrush , most notably specifying the `sslRoot` in one place and passing that down in to builder.

File: lib/certs.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
+/*jslint node: true */
+'use strict';
```
[the remark this answers is not in the document]
Comment (Blackbaud-BobbyEarl) [reply lib/certs.js]: Removed.

File: topics/certs.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+COMING SOON
```
Comment (Blackbaud-TerryHelems) [inline topics/certs.txt]: Should we document this now or can it wait?

File: topics/certs.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+The "skyux certs" command contains several actions around certificate management. Available actions are clean, clone, install, trust, uninstall, and untrust. https://developer.blackbaud.com/skyux/learn/reference/cli-commands/generate
```
Comment (Blackbaud-TerryHelems) [inline topics/certs.txt]: Do we have to include a URL? Can we leave it blank or point to just `cli-commands` until the topic exists?

File: topics/certs.txt (hunk 5 above)
Comment (Blackbaud-BobbyEarl) [reply topics/certs.txt]: Added documentation.

File: topics/certs.txt (hunk 6 above)
Comment (Blackbaud-BobbyEarl) [reply topics/certs.txt]: I updated the URL.  We should be able to get that page created before this releases.  I'll go head and create the initial PR now.

**Comment by Blackbaud-BobbyEarl** [comment]
Closing in favor of https://github.com/blackbaud/skyux-sdk-cli/pull/36

**Pull request closed without merging**